agent/engine/dockerclient/dockerclientfactory_test.go

```diff
@@ -156,64 +154,43 @@ func TestGetClientFailPingNotCached(t *testing.T) {
 	}
 }
 
-func TestFindAvailableVersiosn(t *testing.T) {
+func TestFindSupportedAPIVersions(t *testing.T) {
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
-	mockClient117 := mock_dockeriface.NewMockClient(ctrl)
-	mockClient118 := mock_dockeriface.NewMockClient(ctrl)
-	mockClient119 := mock_dockeriface.NewMockClient(ctrl)
-	mockClient120 := mock_dockeriface.NewMockClient(ctrl)
-	mockClient121 := mock_dockeriface.NewMockClient(ctrl)
-	mockClient122 := mock_dockeriface.NewMockClient(ctrl)
-	mockClient123 := mock_dockeriface.NewMockClient(ctrl)
+	expectedVersions := getAgentVersions()
+	allVersions := getKnownAPIVersions()
+
+	// Set up the mocks and expectations
+	mockClients := make(map[string]*mock_dockeriface.MockClient)
 
-	expectedEndpoint := "expectedEndpoint"
+	for i := 0; i < len(allVersions); i++ {
+		mockClients[string(allVersions[i])] = mock_dockeriface.NewMockClient(ctrl)
+		mockClients[string(allVersions[i])].EXPECT().Ping()
+	}
 
+	// Define the function for the mock client
 	newVersionedClient = func(endpoint, version string) (dockeriface.Client, error) {
 		if endpoint != expectedEndpoint {
 			t.Errorf("Expected endpoint %s but was %s", expectedEndpoint, endpoint)
 		}
 
-		switch DockerVersion(version) {
-		case Version_1_17:
-			return mockClient117, nil
-		case Version_1_18:
-			return mockClient118, nil
-		case Version_1_19:
-			return mockClient119, nil
-		case Version_1_20:
-			return mockClient120, nil
-		case Version_1_21:
-			return mockClient121, nil
-		case Version_1_22:
-			return mockClient122, nil
-		case Version_1_23:
-			return mockClient123, nil
-		default:
-			t.Fatal("Unrecognized version")
+		client, ok := mockClients[version]
+		if !ok {
+			t.Errorf("Client not found for version %s", version)
 		}
-		return nil, fmt.Errorf("This should not happen, update the test")
-	}
 
-	mockClient117.EXPECT().Ping()
-	mockClient118.EXPECT().Ping().Return(fmt.Errorf("Test error!"))
-	mockClient119.EXPECT().Ping()
-	mockClient120.EXPECT().Ping()
-	mockClient121.EXPECT().Ping()
-	mockClient122.EXPECT().Ping()
-	mockClient123.EXPECT().Ping()
-
-	expectedVersions := []DockerVersion{Version_1_17, Version_1_19, Version_1_20, Version_1_21, Version_1_22, Version_1_23}
+		return client, nil
+	}
 
 	factory := NewFactory(expectedEndpoint)
-	versions := factory.FindAvailableVersions()
-	if len(versions) != len(expectedVersions) {
-		t.Errorf("Expected %d versions but got %d", len(expectedVersions), len(versions))
+	actualVersions := factory.FindSupportedAPIVersions()
+	if len(expectedVersions) != len(actualVersions) {
```

Comment:
Can you use `assert.Equal(t, exptectedVersions, actualVersions)` or a variation of it?